File: conventional-commits-common/src/main/java/com/smartling/cc4j/semantic/release/common/ConventionalVersioning.java
package com.smartling.cc4j.semantic.release.common;

import com.smartling.cc4j.semantic.release.common.scm.ScmApiException;

import java.io.IOException;

public interface ConventionalVersioning
{
    LogHandler logHandler();

    SemanticVersionChange getNextVersionChangeType() throws ScmApiException, IOException;

    SemanticVersion getNextVersion(SemanticVersion currentVersion) throws IOException, ScmApiException;
}

File: conventional-commits-common/src/main/java/com/smartling/cc4j/semantic/release/common/ConventionalVersioning.kt
package com.smartling.cc4j.semantic.release.common

import kotlin.Throws
import com.smartling.cc4j.semantic.release.common.scm.ScmApiException
import java.io.IOException

interface ConventionalVersioning {
    fun logHandler(): LogHandler?

    @Throws(ScmApiException::class, IOException::class)
    fun getNextVersionChangeType(): SemanticVersionChange?

    @Throws(IOException::class, ScmApiException::class)
    fun getNextVersion(currentVersion: SemanticVersion): SemanticVersion
}
